cyclonedx/model/component.py

```diff
@@ -766,9 +766,9 @@ def __init__(self, *, name: str, type_: ComponentType = ComponentType.LIBRARY,
                  # Deprecated parameters kept for backwards compatibility
                  namespace: Optional[str] = None, license_str: Optional[str] = None
                  ) -> None:
-        self.type_ = type_
+        self.type = type
         self.mime_type = mime_type
-        if type(bom_ref) == BomRef:
+        if isinstance(bom_ref, BomRef):
```

Comment:
:100: 
and here we have actually a shadowing of the predefined `type()` function.
we could call our variable (not the property) `type_` - this is would be a solution ...
but you found a way around.
`isinstance()` is better anyway, because it is aware of inheritance.

Reply:
Yup! Thankfully our TESTS picked that up 👍